File: javascript/src/TripService.js
'use strict';

let UserSession = require('./UserSession');
let TripDAO = require('./TripDAO');

class TripService {
  getTripsByUser(user, loggedUser) {
    let tripList = [];
    let isFriend = false;

    if (loggedUser === null) {
      throw new Error('User not logged in.');
    }

    return user.isFriendsWith(loggedUser) ? this._findTripsByUser(user) : [];
  }

  _findTripsByUser() {
    return TripDAO.findTripsByUser();
  }
}

module.exports = TripService;

File: java/trip-service-kata/src/main/java/org/craftedsw/tripservicekata/trip/TripService.java
package org.craftedsw.tripservicekata.trip;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import org.craftedsw.tripservicekata.exception.UserNotLoggedInException;
import org.craftedsw.tripservicekata.user.User;
import org.craftedsw.tripservicekata.user.UserSession;

public class TripService {

    public List<Trip> getTripsByUser(User user) throws UserNotLoggedInException {
        User loggedUser = getLoggedUser();
        validate(loggedUser);
        return user.isFriendWith(loggedUser) ? findTripsByUser(user) : Collections.emptyList();
    }


    void validate(User loggedUser) {
        if (loggedUser == null) {
            throw new UserNotLoggedInException();
        }
    }

    List<Trip> findTripsByUser(User user) {
        return TripDAO.findTripsByUser(user);
    }

    User getLoggedUser() {
        return UserSession.getInstance().getLoggedUser();
    }


}
